### ISSUE TITLE
ssh not staying connected after sleep mode

### ISSUE THREAD

**Issue by ghost** [mRemoteNG/mRemoteNG#2296]
## Expected Behavior

when the machine "win 11" goes to sleep mode, when coming back from sleep I find it disconnected
## Current Behavior

the SSH connection should remain connected after sleep mode
## Possible Solution

I checked the 'options' and 'putty session settings' for any option 'keepalive...etc' but didnt find any
## Steps to Reproduce (for bugs)

1. launch mRemoteNG
2. login to any server
3. click "sleep" from the start menu
4. press the button to come back from sleep mode
5. check the connection used in #2

## Context

I need to stay connected after the machine goes to sleep then return
## Your Environment

* Version used: 1.77.1
* Operating System and version (e.g. Windows 10 1709 x64):
* Win 11 64 bit
* the server Linux,OpenSSH_7.4p1

**Comment by simonai1254** [comment]
Hi 

I think this is a limitation of Putty (and the SSH protocol), which is not directly in control of the mRemoteNG project. Eg. I know for sure that putty terminates the sessions when the IP stack changes, so I guess the sleep in Windows 11 also disables the network for power savings and therefore causes the putty sessions to fail.

**Comment by thetuxinator** [comment]
This is normal and expected, this has also todo with session security and in the end every secure remote protocoll should do this to avoid too long session lifetimes. You may use a tool like screen to "keepalive" sessions when connecting back. see https://linux.die.net/man/1/screen

**Comment by Kvarkas** [comment]
Frankly speaking its expected behaviour as mentioned above due security, but may have a quick look a bit later today if its possible to add option keep alive or smth like such, logically it may be option reconnect on wakeup as well.

**Comment by thetuxinator** [comment]
@Kvarkas you can't do anything on the client side actually, because the server has (as of good practice security) some session timeouts and also tcp has them, so it's not woth to spent time on. if the server kills your tcp or ssh session while you are "disconnected/ on standby" -> game over ;-)

**Comment by thetuxinator** [comment]
Ah and i almost forgot, you can't send keepalive when you are offline/in sleep

**Comment by thetuxinator** [comment]
a good explanation can be found here: https://www.quora.com/How-do-you-definitively-keep-SSH-sessions-open-while-putting-a-laptop-to-sleep

**Comment by Kvarkas** [comment]
> @Kvarkas you can't do anything on the client side actually, because the server has (as of good practice security) some session timeouts and also tcp has them, so it's not woth to spent time on. if the server kills your tcp or ssh session while you are "disconnected/ on standby" -> game over ;-)

yes, but who say we cant resend session reconnect as far as have all connection detail? Any caprice for user money :)

**Comment by thetuxinator** [comment]
Well of course you can add some option like a "auto reconnect" in my experience however thats not really present in many tools. the better way is to use some tool like screen where you connect in to the local "screen tool session" when ssh is reconnected. however feel free as you are the boss ;-)

**Comment by Kvarkas** [comment]
hey cmon! I am not a boss (captain perhaps, yes) we are making tool (we - not only who code but also who donate!) that is secure,  easy to use and comfortable, if community wants something and requested this will be implemented... one day... :)

**Comment by thetuxinator** [comment]
Boss was only meant in a funny manner :D :D :D :D

**Comment by thetuxinator** [comment]
As this behaviour is impossible to change/steer on the client-side only (basic network knowledge) and every product has this problem and there are 725 open issues on this project, in my humble opinion this should be closed.

**Comment by journeym** [comment]
So... project has no stable releases in past 3 years, over 700 issues and lead dev is seriously discussing putty sleep circumventing.
Great

**Comment by Kvarkas** [comment]
> So... project has no stable releases in past 3 years, over 700 issues and lead dev is seriously discussing putty sleep circumventing. Great

Always funny how people criticize then don't even donate to the project :)

**Comment by journeym** [comment]
Yeah yeah, keep up the good work mate lol

**Comment by Kvarkas** [comment]
don't worry, I prefer as we say: the quieter you go, the further you'll get

**Comment by thetuxinator** [comment]
Well i was suggesting to better moderate and prioritize the issues with the help of moderators, however if i see, that there are problems discussed if you can avoid a problem that exists by design (TCP/IP -> you have to "kill" a connection after some time -> the same for encrypted protocols, for security reasons) i loose the mind to invest time in this.

**Comment by thetuxinator** [comment]
Btw, just to see if people think logical here, as a test, who will send your "keepalive-packet" when your computer is asleep? :D :D

**Comment by thetuxinator** [comment]
> Frankly speaking its expected behaviour as mentioned above due security, but may have a quick look a bit later today if its possible to add option keep alive or smth like such, logically it may be option reconnect on wakeup as well.

As a last try to get this issue rejected, you want to achieve this:

Client -> Connect to SSH-Server

Open Program A on SSH-Server

Client -> Goto Hibernation/Sleep

Server -> Client is gone for xxx Seconds Closing connection (for security or other reasons (max open connections etc) )

Program A is terminated if not started in background with all proper options or as a daemon or using "screen" or similar tools

Client Wakes Up -> Autoconnect to SSH-Server in a fresh session, Program A is dead

What are you winning?

The only nice solution to the above is to use a software like "screen"

also see: https://www.quora.com/How-do-you-definitively-keep-SSH-sessions-open-while-putting-a-laptop-to-sleep
also see: https://www.tecmint.com/keep-remote-ssh-sessions-running-after-disconnection/

Actually if you get to the point where the SSH-Server would keep the connection open forever (which would make SSH insecure! ) still your Router or ISP would kill the connection after some time and you would not have an existing tcp-session when resumed your Computer.

Hope above explains it enough.

So at the end, this is not a problem, this is by design, so "dontfix/reject" makes sense here.

**Comment by robertpopa22** [comment]
Triage refresh (P2): this issue has been inactive for a long time. Please retest on the latest build and share reproducible steps, expected vs actual behavior, and logs/screenshots. If there is no confirmation/update, maintainers can close it as stale or not reproducible.

**Comment by robertpopa22** [comment]
**Fix available for testing**

A fix for this issue has been implemented in our community fork (`robertpopa22/mRemoteNG`, branch `main`). A beta build including this fix is available from the fork's Releases page.

Please test and report if this resolves your issue.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*
